src/containers/my-courses/course-search-tool-bar/CourseSearchToolbar.tsx

```diff
@@ -0,0 +1,145 @@
+  resetPage: () => void
+}
+
+const levelLists = Object.values(ProficiencyLevelEnum)
+
+const CourseSearchToolbar = ({
+  filters,
+  resetPage,
+  filterActions
+}: OfferSearchToolbarProps) => {
+  const { t } = useTranslation()
+  const { isMobile } = useBreakpoints()
+  const { updateFilterInQuery } = filterActions
+
+  const getSubjectsNames = useCallback(
+    () => subjectService.getSubjectsNames(filters.categoryId),
+    [filters.categoryId]
+  )
+  const onCategoryChange = (
+    _: SyntheticEvent,
+    value: CategoryNameInterface | null
+  ) => {
+    updateFilterInQuery(value?._id ?? '', 'categoryId')
+    updateFilterInQuery('', 'subjectId')
+    resetPage()
+  }
+
+  const onSubjectChange = (
+    _: SyntheticEvent,
+    value: SubjectNameInterface | null
+  ) => {
+    updateFilterInQuery(value?._id ?? '', 'subjectId')
+    resetPage()
+  }
+
+  const [selectedLevel, setSelectedLevel] = useState<ProficiencyLevelEnum[]>([])
+
+  const onLevelChange = (event: SelectChangeEvent<ProficiencyLevelEnum[]>) => {
+    const {
+      target: { value }
+    } = event
+    setSelectedLevel(value as ProficiencyLevelEnum[])
+    updateFilterInQuery(value as ProficiencyLevelEnum[], 'proficiencyLevel')
+    resetPage()
+  }
+
+  const AppAutoCompleteList = (
+    <>
+      <AsyncAutocomplete
+        labelField='name'
+        onChange={onCategoryChange}
+        service={categoryService.getCategoriesNames}
+        sx={styles.autocomplete}
+        textFieldProps={{
+          label: t('breadCrumbs.categories'),
+          helperText: t('myCoursesPage.filterLabel.categories')
+        }}
+        value={filters.categoryId}
+        valueField='_id'
+      />
+      <AsyncAutocomplete
+        disabled={!filters.categoryId}
+        labelField='name'
+        onChange={onSubjectChange}
+        service={getSubjectsNames}
+        sx={styles.autocomplete}
+        textFieldProps={{
+          label: t('breadCrumbs.subjects'),
+          helperText: t('myCoursesPage.filterLabel.subjects')
+        }}
+        value={filters.subjectId}
+        valueField='_id'
+      />
+      <FormControl>
+        <InputLabel id='demo-multiple-checkbox-label'>Levels</InputLabel>
+        <Select
+          MenuProps={styles.menuProps}
+          id='demo-multiple-checkbox'
+          input={<OutlinedInput label='Level' />}
+          labelId='demo-multiple-checkbox-label'
+          multiple
+          onChange={onLevelChange}
+          renderValue={(selected) => {
+            return selected.join(', ')
+          }}
+          style={styles.drowlevel as CSSProperties}
+          value={filters.proficiencyLevel}
+        >
+          {levelLists.map((item) => (
+            <MenuItem key={item} value={item}>
+              <Checkbox checked={selectedLevel.indexOf(item) > -1} />
+              <ListItemText primary={item} />
+            </MenuItem>
+          ))}
+        </Select>
+        <FormHelperText sx={styles.drowstyle as CSSProperties}>
```

Comment:
Do you really need this as CSSProperties here and a few lines below?

Reply:
yes, to avoid typescript warnings